test/integration.test.ts

```diff
@@ -474,3 +484,153 @@ describe('dleq', () => {
+		const restAmount = sumProofs(meltKeep) - wallet.getFeesForProofs(meltKeep);
+		// First we unlock all the proofs that we have left
+		const unlockedProofs = await wallet.send(restAmount, meltKeep, {
+			privkey: hexSk
+		});
+		// Just to receive them and lock them again, but this time overwriting the default factory
+		const newProofs = await wallet.receive(
+			{ proofs: unlockedProofs.send, mint: mintUrl },
+			{ outputData: (a, k) => OutputData.createSingleP2PKData({ pubkey: 'testKey' }, a, k.id) }
+		);
+		// Our factory also applies to the receive method, so we expect all received proofs to be locked
+		expectProofsSecretToEqual(newProofs, 'testKey');
+	}, 15000);
+	test('Manual Factory Mint', async () => {
+		function createFactory(pubkey: string): OutputDataFactory {
+			function inner(a: number, k: MintKeys) {
+				return OutputData.createSingleP2PKData({ pubkey: pubkey }, a, k.id);
+			}
+			return inner;
+		}
+
+		const mint = new CashuMint(mintUrl);
+		const wallet = new CashuWallet(mint);
+
+		const quote = await wallet.createMintQuote(21);
+		await new Promise((res) => setTimeout(res, 1000));
+		const proofs = await wallet.mintProofs(21, quote.quote, {
+			outputData: createFactory('mintTest')
+		});
+		expectProofsSecretToEqual(proofs, 'mintTest');
+	});
+	test('Manual Factory Send', async () => {
+		function createFactory(pubkey: string): OutputDataFactory {
+			function inner(a: number, k: MintKeys) {
+				return OutputData.createSingleP2PKData({ pubkey }, a, k.id);
+			}
+			return inner;
+		}
+
+		const mint = new CashuMint(mintUrl);
+		const wallet = new CashuWallet(mint);
+
+		const quote = await wallet.createMintQuote(21);
+		await new Promise((res) => setTimeout(res, 1000));
+		const proofs = await wallet.mintProofs(21, quote.quote);
+		const amount = sumProofs(proofs) - wallet.getFeesForProofs(proofs);
+		const { send, keep } = await wallet.send(amount, proofs, {
+			outputData: { send: createFactory('send'), keep: createFactory('keep') }
+		});
+		expectProofsSecretToEqual(send, 'send');
+		expectProofsSecretToEqual(keep, 'keep');
+	});
+	test('Manual BlindingData', async () => {
+		const mint = new CashuMint(mintUrl);
+		const wallet = new CashuWallet(mint);
+		const keys = await wallet.getKeys();
+
+		const quote = await wallet.createMintQuote(40);
+		await new Promise((res) => setTimeout(res, 1000));
+		const proofs = await wallet.mintProofs(40, quote.quote);
+		const data1 = OutputData.createP2PKData({ pubkey: 'key1' }, 10, keys);
+		const data2 = OutputData.createP2PKData({ pubkey: 'key2' }, 10, keys);
+		const { keep, send } = await wallet.send(20, proofs, {
+			outputData: { send: [...data1, ...data2] }
+		});
+		const key1Sends = send.slice(0, data1.length);
+		const key2Sends = send.slice(data1.length);
+		expectProofsSecretToEqual(key1Sends, 'key1');
+		expectProofsSecretToEqual(key2Sends, 'key2');
+	});
+});
+describe('Keep Vector and Reordering', () => {
```

Comment:
it doesn't seem like you are testing the "Keep Vector" here, just that `send` is properly created when `keepAmounts` is provided

Reply:
It does test the vector, but it is hidden behind the scenes. Any outputs in swap-payload ssend to the mint are now ordered by amount, to increase privacy. By testing that the output of `receive` is in the same order as the keepAmounts specified, we proof that the reordering of proofs works as expected. The code that does this can be found in CashuWallet.ts#305 (reordering) CashuWallet.ts#872 (sorting)